src/pages/Earn/index.tsx

```diff
@@ -140,18 +152,22 @@ export default function Earn({
         <PoolSection>
           {stakingRewardsExist && stakingInfos?.length === 0 ? (
             <Loader style={{ margin: 'auto' }} />
-          ) : !stakingRewardsExist || stakingInfoResults?.length === 0 ? (
+          ) : !stakingRewardsExist || stakingInfoCards?.length === 0 ? (
             t('earnPage.noActiveRewards')
           ) : (
-            stakingInfoResults?.map(stakingInfo => (
-              <PoolCard
-                apr={stakingInfo.apr}
-                key={stakingInfo.stakingRewardAddress}
-                stakingInfo={stakingInfo}
-                migration={MIGRATIONS.find(migration => migration.from.stakingRewardAddress === stakingInfo.stakingRewardAddress)?.to}
-                version={version}
+            <>
+              <SearchInput
+                type="text"
+                id="token-search-input"
+                placeholder={t('searchModal.tokenName')}
+                value={searchQuery}
+                onChange={e => setSearchQuery(e.target.value.trim().toUpperCase())}
               />
-            ))
+              {stakingInfoCards?.filter(card =>
+                  card.props.stakingInfo.tokens[0].symbol.toUpperCase().includes(searchQuery)
+                || card.props.stakingInfo.tokens[1].symbol.toUpperCase().includes(searchQuery))
+              }
```

Comment:
though this is looking good and it will work but here is my 2 cents

1. we should not use props like this `card.props`.  
2. we should do the filter in `useEffect(() => {}, [searchQuery])`, so filter will only be done whenever `searchQuery` otherwise filter will run whenever any other states got changed
3. the changes you done in above `useEffect` from `setStakingInfoResults(results)` to `setStakingInfoCards(poolCards)` will not make any difference as long as `PoolCard` has `key` and is pure component.

let me know your thoghts

Reply:
Great! I'm not super happy with the props filtering either, but addressed the rest of the concerns and we can iterate on a better/efficient way to filter the cards